packages/devtools_app/lib/src/profiler/cpu_profile_model.dart

```diff
@@ -172,6 +173,92 @@ class CpuProfileData {
     );
   }
 
+  factory CpuProfileData.filterFrom(
+    CpuProfileData originalData,
+    bool Function(CpuStackFrame) includeFilter,
+  ) {
+    final filteredCpuSamples = <CpuSample>[];
+    void includeSampleOrWalkUp(
+      CpuSample sample,
+      Map<String, Object> sampleJson,
+      CpuStackFrame stackFrame,
+    ) {
+      if (includeFilter(stackFrame)) {
+        // Modify the name value for this sample trace event to reflect the
+        // leaf most node for the filtered sample.
+        sampleJson[stackFrameIdKey] = stackFrame.id;
```

Comment:
Can we avoid modifying the json? Rather than having to copy the json it might be clearer and faster to optionally pass in a stackFrameIdOverride to CpuSample.

Reply:
done. moved this modification into a toJson getter so that exports of the filtered CPU profile data are accurate